### PR TITLE
CI, Tests, and CMake

### PR DESCRIPTION
**Pull request by kgerheiser**
Changes from the release branch into develop, CI, and pfunit tests.

### CODE REVIEW

**Comment by GeorgeVandenberghe-NOAA** [comment]
Clarify instructions to clone, build and test this.   I can supply my own bacio for the tests if a full dependency test is too difficult

**Review by aerorahul — APPROVED** [review]
I was able to clone and build this with bacio.
Since I don't have pfUnit, I did not test the tests.

**Comment by aerorahul** [comment]
> Clarify instructions to clone, build and test this. I can supply my own bacio for the tests if a full dependency test is too difficult
@GeorgeVandenberghe-NOAA 
Load your favorite compiler, GNU, Clang, Intel e.g. shows gnu below
```
export CC=gcc
export FC=gfortran

git clone https://github.com/noaa-emc/nceplibs-bacio
cd nceplibs-bacio
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=$PWD/../install -DBUILD_SHARED_LIBS=ON ..
make 
make install

cd ../..
git clone https://github.com/kgerheiser/nceplibs-sigio
cd nceplibs-sigio
git checkout feature/tests
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=$PWD/../install -DBUILD_SHARED_LIBS=ON -Dbacio_DIR=../../nceplibs-bacio/install/lib/cmake/bacio ..
make
make install
```

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -10,20 +10,17 @@ elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
 endif()
 
 set(fortran_src sigio_module.f sigio_r_module.f)
-set(kind 4)
 
 set(lib_name ${PROJECT_NAME}_4)
```
Comment (aerorahul) [inline src/CMakeLists.txt]: If we are building in real32 only, why is there a need for `_4`?

Comment (kgerheiser) [reply src/CMakeLists.txt]: There's not, that line was removed

Comment (kgerheiser) [reply src/CMakeLists.txt]: Or do you mean the rest of the `_4` out there? I don't know, I just left it the way it was.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS
+      "-g -fbacktrace -ffree-form -fconvert=big-endian -funroll-loops")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
+endif()
+
+set(fortran_src sigio_module.f sigio_r_module.f)
+
+set(lib_name ${PROJECT_NAME}_4)
```
Comment (aerorahul) [inline src/CMakeLists.txt]: L14 of `src/CMakeLists.txt` in your branch is:
```
set(lib_name ${PROJECT_NAME}_4)
```

File: src/CMakeLists.txt (hunk 1 above)
Comment (aerorahul) [reply src/CMakeLists.txt]: well, we are not building real64, so gut the distinction.

**Pull request merged**